### PR TITLE
Adds unique_id field to source configurations and fixes pagination.

### PR DESCRIPTION
**Pull request by caiotomazelli**

### CODE REVIEW

**Review by blackwire — DISMISSED** [review]
This looks good! Like the idea of providing information around what it defaults to that will show up in the UI. Good call.

File: dags/sources/bigquery.py [hunk 1]
```diff
@@ -59,20 +64,24 @@ def get_data(
...
 
-    rows = []
-    for element in query_job.result():
-      # create dict to hold results and respect the return type
-      row = {}
-      for f in fields:
-        if f in element.keys():
-          row[f] = element.get(f)
-      rows.append(row)
+    try:
```
Comment (blackwire) [inline dags/sources/bigquery.py:72]: Why does the loop require a "BadRequest" try/catch, but the self.client.query doesn't? Was there a special situation that occurs with BigQuery client here?

File: dags/sources/bigquery.py [hunk 2]
```diff
@@ -84,7 +93,10 @@ def schema() -> Optional[ProtocolSchema]:
                 validation="^[a-zA-Z0-9_]{1,1024}$")),
             ("table", str, Field(
                 description="The name of your BigQuery table.",)),
-            
+            ("unique_id", Optional[str], Field(
+                description=f"Unique id column name to be used by BigQuery. Defaults to '{_UNIQUE_ID_DEFAULT_NAME}' when nothing is provided.",
+                default=_UNIQUE_ID_DEFAULT_NAME
```
Comment (blackwire) [inline dags/sources/bigquery.py:98]: Nit: on the credentials the default value is first and on this one it's second.

Comment (caiotomazelli) [reply dags/sources/bigquery.py:98]: Nice catch

File: dags/sources/bigquery.py (hunk 1 above)
Comment (caiotomazelli) [reply dags/sources/bigquery.py:72]: The actual call to BQ is done upon calling query_job.result().

But I see that this is not clear the way it is written, I'll add a explicit variable to make it obvious.

BadRequest is being throw when one tries to ORDER BY using a field that does not exist in the table. I'll also add a comment to that.

**Review by blackwire — APPROVED** [review]

**Pull request merged**